www_src/pages/page/page.jsx

```diff
@@ -164,6 +166,14 @@ var Page = React.createClass({
           state.currentElementId = id;
         }
         this.setState(state);
+
+        //Ensure we don't reach a race conditions where the buttons are
+        //re-enabled before the menu is hidden. 200ms matches the css animation
+        //speed.
+
+        setTimeout(function() {
```

Comment:
this should not have been here. This should not be a timeout, but should have been triggered by the child element being done doing what it's doing, informing its parent that it's now done and the buttons can be shown again. In traditional HTML+JS you would use a timeout, but in React, using a timeout is a sign that lifecycles are not being respected.